unreleased_history/bug_fixes/block_align_checksum_mismatch.md

```diff
@@ -0,0 +1 @@
+Fix a bug causing `VerifyFileChecksums()` to return false-positive corruption under `BlockBasedTableOptions::block_align=true`
```

Comment:
Already written files will continue failing VerifyFileChecksums() because this is stored in the MANIFEST, right?

Reply:
yeah

Reply:
It feels more like the fix is to stop writing files with wrong checksums, then, since the false positive reports would continue beyond this bug fix